tools/src/test/java/org/apache/kafka/tools/reassign/ReassignPartitionsUnitTest.java

```diff
@@ -300,6 +300,12 @@ public void testGetReplicaAssignments() throws Exception {
 
             assertEquals(assignments,
                 getReplicaAssignmentForPartitions(adminClient, new HashSet<>(asList(new TopicPartition("foo", 0), new TopicPartition("bar", 0)))));
+
+            assignments.clear();
+
+            UnknownTopicOrPartitionException exception = assertThrows(UnknownTopicOrPartitionException.class,
+                () -> getReplicaAssignmentForPartitions(adminClient, new HashSet<>(asList(new TopicPartition("foo", 0), new TopicPartition("foo", 10)))));
+            assertEquals("Unable to find partition: foo-10", exception.getMessage());
```

Comment:
If we change it to ExecutionException, should we change this? Ex:
```
Exception  exception = assertThrows(ExecutionException.class,
                () -> getReplicaAssignmentForPartitions(adminClient, new HashSet<>(asList(new TopicPartition("foo", 0), new TopicPartition("foo", 10)))));
Exception causeException = exception.getCause();
AssertTrue(causeException instanceof UnknownTopicOrPartitionException);
assertEquals("Unable to find partition: foo-10", causeException.getMessage());
```

Reply:
Oops, my bad...